### PR TITLE
chore: Angular 18 support

### PR DESCRIPTION
**Pull request by lukaVarga**

### CODE REVIEW

File: test-angular-versions/v18/src/app/app.config.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+export const states = [
+  { name: 'home', url: '/home', component: HomeComponent },
+  { name: 'about', url: '/about', component: AboutComponent },
+  { name: 'lazy.**', url: '/lazy', loadChildren: () => import('./lazy/lazy.module').then((m) => m.LazyModule) },
+];
+
+export function config(router: UIRouter) {
+  router.urlService.rules.initial({ state: 'home' });
+}
```
Comment (sbschorer) [inline test-angular-versions/v18/src/app/app.config.ts:16]: This seems to be duplicated from `app.module.ts`

File: downstream_projects.json [hunk 2]
```diff
@@ -1,8 +1,8 @@
 {
   "packageDir": "./dist",
   "projects": {
-    "sample-app-angular": "https://github.com/ui-router/sample-app-angular.git",
-    "angular17": "./test-angular-versions/v17",
-    "typescript52": "./test-typescript-versions/typescript5.2"
+    "sample-app-angular": "https://github.com/lukaVarga/ui-router-sample-app-angular.git",
```
Comment (sbschorer) [inline downstream_projects.json:4]: I guess, this should be changed back to
```suggestion
    "sample-app-angular": "https://github.com/ui-router/sample-app-angular.git",
```

**Review by sbschorer — CHANGES_REQUESTED** [review]
Thanks for opening this PR :)

**Comment by lukaVarga** [comment]
Thanks for the review. Adjusted :)

**Pull request merged**